compiler/test/dotty/tools/dotc/CompilationTests.scala

```diff
@@ -96,7 +96,8 @@ class CompilationTests extends ParallelTesting {
       "tests/neg-custom-args/fatal-warnings/xfatalWarnings.scala",
       defaultOptions.and("-nowarn", "-Xfatal-warnings")
     ) +
-    compileFile("tests/pos-special/typeclass-scaling.scala", defaultOptions.and("-Xmax-inlines", "40"))
+    compileFile("tests/pos-special/typeclass-scaling.scala", defaultOptions.and("-Xmax-inlines", "40")) +
+    compileDir("tests/pos-special/i6152", defaultOptions, separateCompilation = true)
```

Comment:
Why use pos-special ? If you put i6152 in tests/pos it should work (except that decompilation tests are broken when java files are involved, we work around this by putting the tests in tests/pos-java-interop)

Reply:
Indeed it worked. It seems `compileFilesInDir` does not treat the sources in its base directory as the part of the same compilation, which confused me. Makes sense.